ci_test/install_agent_7_macos.yaml

```diff
@@ -0,0 +1,29 @@
+---
+
+- hosts: all
+  roles:
+    - { role: '/Users/distiller/project/' }
+  vars:
+    datadog_api_key: "11111111111111111111111111111111"
+    datadog_user: "distiller"
```

Comment:
I am slightly concerned about the `datadog_user` property. It is used extensively in the linux and windows code to set the owner of datadog files, etc. The default value is of course not available on the circleci macs or my orgs machines so without setting this `datadog_user` property the install does not function correctly. Maybe I am misunderstanding something? I am worried about user error if this property is not specified.

Reply:
Maybe the intention is that outside of this playbook the user must create the `dd-agent` user or specify an override. I don't believe it should be the responsibility of this playbook to create users as that could have significant consequences on the running machines.

Reply:
The easiest solution I can think of is to move the `.plist` file created by the Datadog Agent installer to the root folder `/Library/LaunchAgents/` - but I am wondering if that would be acceptable to the Datadog team.

Reply:
i was able to resolve these issues by not using `community.general.launchd` and calling launchctl commands directly

Reply:
Yes, I think doing a systemwide LaunchDaemon installation (== putting plist files in `/Library/LaunchAgents`) is the way to go here. Do note that I actually have a PR under development for our `install_mac_os.sh` script that does just that to enable system-wide installation without the user logged in via gui: https://github.com/DataDog/datadog-agent/pull/10621 - it'd probably make sense to make my PR and your PR behave consistently, so I'll do my review in that spirit.

Reply:
I was familiarizing myself with the `launchctl` options and noted that Apple considers many of the available commands deprecated. You will see I chose to use the non-deprecated command forms that include `domain/uid` such as `launchctl bootstrap gui/<uid> <plist-file>`. There are other domains which seem better suited to automation (system, user) but I was not able to get the plist file included with datadog agent to load in those domains. I am not familiar enough with macOS details like this to understand what is happening. Unfortunately the `gui` domain requires the user to have a logged in gui session on the target machine.

Reply:
My PR (mentioned above) that I did for `install_mac_os.sh` actually does what you're trying to achieve, I think. It moves the plist file from user's homedir to `/Library/LaunchDaemons` and adds a UID/GID entry so that the Agent runs under desired user/group. Have you tried that? Feel free to take some inspiration from my PR might help.

I think it would also make a lot of sense to create `datadog_macos_user`/`datadog_macos_group` variables and run the agent under these to ensure things are configurable.